include/zephyr/drivers/video.h

```diff
@@ -140,6 +140,7 @@ enum video_endpoint_id {
  * Identify video event.
  */
 enum video_signal_result {
+	VIDEO_BUF_STARTED,
 	VIDEO_BUF_DONE,
 	VIDEO_BUF_ABORTED,
 	VIDEO_BUF_ERROR,
```

Comment:
@loicpoulain I was wondering this too. Maybe these are different:

During the hardware interrupt, the driver would pick the next buffer and submit it.

The hardware interrupt would be "Start of Frame": it is not yet possible to change the configuration as the buffer is not enqueued yet to the hardware.

Once the buffer is enqueued, the "Buffer processing started" event would happen: it is now possible, for hardware supporting it, to change the configuration (i.e. different pitch/width/height for the next frame, i.e. FPGA alternating frames across various sources).

If the driver waits for more buffers to free-up, there could be just a few us or minutes (i.e. stream on pause) between these two events.

@epc-ake did I understand your use-case well (as discussed on #video on Discord)?

```suggestion
	/** Event raised by the Start of Frame interrupt */
	VIDEO_START_OF_FRAME,
	/** After the SoF event, the driver acquired another buffer and successfully submitted it to the hardware */
	VIDEO_BUF_STARTED,
	/** Event raised by the End of Frame interrupt */
	VIDEO_END_OF_FRAME,
	/** After the EoF event, the driver released another buffer that can be dequeued */
	VIDEO_BUF_DONE,
	/** An ongoing transfer was aborted mid-way */
	VIDEO_BUF_ABORTED,
	/** An error occurred while attempting to process the buffer */
	VIDEO_BUF_ERROR,
```

Something like this?

Reply:
I tried to put it on paper. 
Not sure how well I did but hopefully it makes my idea clear ;-)
![Screenshot from 2024-09-17 16-43-55](https://github.com/user-attachments/assets/73f1d529-9036-440f-ba3d-8ba426354214)

Reply:
Thank you I had the wrong model in mind!

It is usually too late to do any enqueuing from the IRQ signal: the buffers need to already be submitted to the hardware before the SOF interrupt happens. The `VIDEO_BUF_STARTED` and `VIDEO_START_OF_FRAME` are then happening at the same moment.

Second attempt:

```suggestion
	/** Event raised by the Start of Frame interrupt, unconditionally */
	VIDEO_START_OF_FRAME,
	/** Event raised by the Start of Frame interrupt, if a buffer was submitted */
	VIDEO_BUF_STARTED,
	/** Event raised by the End of Frame interrupt, unconditinoally */
	VIDEO_END_OF_FRAME,
	/** Event raised by the End of Frame interrupt, if a buffer was submitted */
	VIDEO_BUF_DONE,
	/** Event raised if an ongoing transfer was aborted before completion */
	VIDEO_BUF_ABORTED,
	/** Event raised if an error happened after submitting a buffer */
	VIDEO_BUF_ERROR,
```